packages/block-library/src/latest-posts/edit.js

```diff
@@ -156,14 +156,14 @@ export default function LatestPostsEdit( { attributes, setAttributes } ) {
 
 	// If a user clicks to a link prevent redirection and show a warning.
 	const { createWarningNotice, removeNotice } = useDispatch( noticeStore );
-	let noticeId;
+	const noticeIdRef = useRef();
 	const showRedirectionPreventedNotice = ( event ) => {
 		event.preventDefault();
 		// Remove previous warning if any, to show one at a time per block.
-		removeNotice( noticeId );
-		noticeId = `block-library/core/latest-posts/redirection-prevented/${ instanceId }`;
+		removeNotice( noticeIdRef.current );
```

Comment:
Seems that tracking the notice ID and the `removeNotice` call is a pointless exercise 🙂 When I'm looking at what the [reducer is doing](https://github.com/WordPress/gutenberg/blob/trunk/packages/notices/src/store/reducer.js#L17-L22) with the `CREATE_NOTICE` action, it removes the old notice with the same ID automatically! We don't need to do it explicitly.

Reply:
I think that would be a better approach. Resolving reassignment error introduced know flaky error, so we still end up in with an error in the same file 😅 

> Reassigning a variable after render has completed can cause inconsistent behavior on subsequent renders. Consider using state instead  react-compiler/react-compiler

Reply:
Ah, what a mess. Let me follow-up once more 😅

Reply:
See:
- #66409